src/main/drivers/adc_impl.h

```diff
@@ -73,8 +73,10 @@ typedef struct adcTagMap_s {
 #define ADC_DEVICES_345 ((1 << ADCDEV_3)|(1 << ADCDEV_4)|(1 << ADCDEV_5))
 
 typedef struct adcDevice_s {
+#if !defined(SIMULATOR_BUILD)
     ADC_TypeDef* ADCx;
```

Comment:
Perhaps #define ADC_TYPE_DEF in platform_mcu.h then check for definition before use?

Reply:
It is possible. 
I tried to get includes fixed, sometimes in quickest way possible. But maybe it can be fixed as PICO refactoring